stl/src/print.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// print.cpp -- C++23 <print> implementation
+
+#include <__msvc_print.hpp>
+#include <cstdio>
+#include <expected>
+#include <internal_shared.h>
+#include <io.h>
+#include <limits>
+#include <stdexcept>
+#include <string>
+#include <string_view>
+#include <type_traits>
+
+#include <Windows.h>
+
+namespace {
+    [[nodiscard]] __std_unicode_console_retrieval_result __stdcall _Get_console_handle_from_file_stream(
+        FILE* const _File_stream) noexcept {
+        const int _Fd = _fileno(_File_stream);

```

Comment:
Putting this on this line but unrelated:

I don't like using the `blah{.foo = ...}` syntax; I'd prefer doing something along the lines of:

```cxx
            __std_unicode_console_retrieval_result _Result;
            ...
            _Result._Error = __std_win_error::_Not_supported;
            return _Result;
```

Reply:
What's wrong with the designated initializers, is it just personal dislike?
I personally find this as one of the most beautiful addition to `C++20` 😎

Reply:
Having conferred with other maintainers, I have been told I'm just being a fuddy-duddy; resolving.